src/plugins/axes/AxesConfig.cc

```diff
@@ -0,0 +1,379 @@
+      << "]. Origin axes plugin won't work." << std::endl;
+    return;
+  }
+
+  if (engine->SceneCount() == 0)
+    return;
+
+  // assume there is only one scene
+  // load scene
+  this->dataPtr->scene = engine->SceneByIndex(0);
+  if (!this->dataPtr->scene)
+  {
+    ignerr << "Internal error: scene is null." << std::endl;
+    return;
+  }
+
+  if (!this->dataPtr->scene->IsInitialized() ||
+    this->dataPtr->scene->VisualCount() == 0)
+  {
+    return;
+  }
+
+  EntitiesInScene();
+}
+
+/////////////////////////////////////////////////
+bool AxesConfig::eventFilter(QObject *_obj, QEvent *_event)
+{
+  if (_event->type() == ignition::gui::events::Render::kType)
+  {
+    // This event is called in Scene3d's RenderThread, so it's safe to make
+    // rendering calls here
+    UpdateOriginArrows();
+  }
+
+  // Standard event processing
+  return QObject::eventFilter(_obj, _event);
+}
+
+/////////////////////////////////////////////////
+void AxesConfig::UpdateActiveAxes()
+{
+  for (auto vis : this->dataPtr->activeAxesMap)
+  {
+    auto visAxes = std::dynamic_pointer_cast<rendering::AxisVisual>(
+          this->dataPtr->scene->VisualByName(vis.first + "Axes"));
+
+    auto visEntity = std::dynamic_pointer_cast<rendering::Visual>(
+          this->dataPtr->scene->VisualByName(vis.first));
+
+    if (visEntity && visAxes)
+    {
+      math::Pose3d pose_entity = visEntity->LocalPose();
+      visAxes->SetLocalPose(vis.second + pose_entity);
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+void AxesConfig::UpdateOriginArrows()
+{
+  if (this->dataPtr->name_axes.empty())
+    return;
+
+  // Load axes if they don't already exist
+  this->LoadAxesbyName(this->dataPtr->name_axes);
+
+  // If axes were not loaded successfully, don't update
+  if (!this->dataPtr->axes)
+    return;
+
+  // Updating the poses
+  this->UpdateActiveAxes();
+
+  if (!this->dataPtr->dirty)
+    return;
+
+  // Save the axesVisual in the structure if it doesn't exist or
+  // update the pose
+  auto it = this->dataPtr->activeAxesMap.find(this->dataPtr->name_axes);
+  if (it == this->dataPtr->activeAxesMap.end())
+  {
+    this->dataPtr->activeAxesMap.insert(
+      std::pair<std::string, math::Pose3d>(
+        this->dataPtr->name_axes,
+        this->dataPtr->pose));
+  }
+  else
+  {
+    it->second = this->dataPtr->pose;
+  }
+
+  // update visibility
+  this->dataPtr->axes->SetVisible(this->dataPtr->visible);
+  // Update type: arrow or line
+  if (this->dataPtr->visible)
+  {
+    this->dataPtr->axes->ShowAxisHead(this->dataPtr->isArrow);
+  }
+  // update scale
+  this->dataPtr->axes->SetLocalScale(1, 1, this->dataPtr->length * 2);
+
+  this->dataPtr->dirty = false;
+}
+
+/////////////////////////////////////////////////
+const QStringList AxesConfig::comboList()
+{
+  return itemComboList;
+}
+
+/////////////////////////////////////////////////
+void AxesConfig::SetComboList(const QStringList &comboList)
+{
+  if (itemComboList != comboList)
```

Comment:
```suggestion
  if (this->itemComboList != comboList)
```

The same everywhere else for this variable